xapian-core/cluster/cluster.cc

```diff
@@ -304,6 +311,11 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	double tf = 1 + log((double)wdf);
 	double idf = log(size / termfreq);
 	double wt = tf * idf;
+
+	// If weight of a term is zero, it doesn't contribute to the termlist
+	if (wt == 0)
+	    continue;
```

Comment:
> Is termfreq > size possible? That would lead to negative wt which seems undesirable, so if it can happen we should probably exclude that too.

That doesn't seem like a possible case because the maximum termfreq of any term will be equal to the number of documents according to how it is being calculated. But I guess it's a common case for a few words to have size ==  termfreq (excluding stopwords). These could be terms that are common in context of the corpus (sports related documents could contain the word 'play' in all documents). So it looks useful to remove these terms before calculating their weight.